controllers/etcd_controller.go

```diff
@@ -379,10 +377,6 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 		return ctrl.Result{
 			Requeue: true,
 		}, err
-	} else if waitForStatefulSetCleanup {
```

Comment:
stsDeployer.Destroy currently does not wait. Should this also call `WaitCleanup`? Maybe use `gardenercomponent.OpWaiter`?

Reply:
Nice catch 👍